drivers/SmartThings/zwave-garage-door-opener/src/test/test_ecolink_garage_door_operator.lua

```diff
@@ -0,0 +1,320 @@
+-- Copyright 2022 SmartThings
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+local test = require "integration_test"
+local capabilities = require "st.capabilities"
+local zw = require "st.zwave"
+local zw_test_utils = require "integration_test.zwave_test_utils"
+local BarrierOperator = (require "st.zwave.CommandClass.BarrierOperator")({ version = 1 })
+local SensorMultilevel = (require "st.zwave.CommandClass.SensorMultilevel")({ version = 11 })
+local t_utils = require "integration_test.utils"
+local Configuration = (require "st.zwave.CommandClass.Configuration")({ version = 4 })
+
+
+local garage_door_endpoints = {
+  {
+    command_classes = {
+      {value = zw.BARRIER_OPERATOR},
+      {value = zw.CONFIGURATION},
+      {value = zw.SENSOR_MULTILEVEL},
+      {value = zw.NOTIFICATION}
+    }
+  }
+}
+
+local mock_garage_door = test.mock_device.build_test_zwave_device({
+  profile = t_utils.get_profile_definition("eco-zw-gdo-profile.yml"),
+  zwave_endpoints = garage_door_endpoints,
+  zwave_manufacturer_id = 0x014A,
+  zwave_product_type = 0x0007,
+  zwave_product_id = 0x4731,
+})
+
+local function test_init()
+  test.mock_device.add_test_device(mock_garage_door)
+end
+
+test.set_test_init_function(test_init)
+
+test.register_message_test(
+    "BarrierOperator reports value 0x00 should be handled as proper capabilities",
+    {
+      {
+        channel = "zwave",
+        direction = "receive",
+        message = { mock_garage_door.id, zw_test_utils.zwave_test_build_receive_command(BarrierOperator:Report({ value = 0x00 })) }
+      },
+      {
+        channel = "capability",
+        direction = "send",
+        message = mock_garage_door:generate_test_message("main", capabilities.doorControl.door.closed())
+      },
+      {
+        channel = "capability",
+        direction = "send",
+        message = mock_garage_door:generate_test_message("contactSensor", capabilities.contactSensor.contact.closed())
+      }
+    }
+)
+
+test.register_message_test(
+        "BarrierOperator set value 0xFF should be handled as proper capabilities",
+        {
+            {
+                channel = "zwave",
+                direction = "receive",
+                message = { mock_garage_door.id, zw_test_utils.zwave_test_build_receive_command(BarrierOperator:Report({ value = 0xFF })) }
```

Comment:
the field is called `state` in the BarrierOperator Report

Reply:
Fixed.